test/test_propagate_filter.py

```diff
@@ -101,59 +124,104 @@ def test_IIRuncFilter():
+
+    # compare lengths
+    assert len(y) == len(input_signal["x"])
+    assert len(Uy) == len(input_signal["Ux"])
+
+
+def test_IIRuncFilter_shape_in_chunks(
+    input_signal, run_IIRuncFilter_in_chunks
+):
+    y, Uy = run_IIRuncFilter_in_chunks
+
+    # compare lengths
+    assert len(y) == len(input_signal["x"])
+    assert len(Uy) == len(input_signal["Ux"])
+
+
+def test_IIRuncFilter_identity_nonchunk_chunk(
+    run_IIRuncFilter_all_at_once, run_IIRuncFilter_in_chunks
+):
+
+    y1, Uy1 = run_IIRuncFilter_all_at_once
+    y2, Uy2 = run_IIRuncFilter_in_chunks
 
     # check if both ways of calling IIRuncFilter yield the same result
-    assert np.allclose(yb, y)
-    assert np.allclose(Uyb, Uy)
+    assert np.allclose(y1, y2)
+    assert np.allclose(Uy1, Uy2)
 
+@pytest.mark.parametrize("kind", ["diag", "corr"])
+def test_FIR_IIR_identity(kind, fir_filter, input_signal):
 
-def test_FIR_IIR_identity():
+    # run signal through both implementations
+    y_iir, Uy_iir, _ = IIRuncFilter(*input_signal.values(), **fir_filter, kind=kind)
+    y_fir, Uy_fir = FIRuncFilter(
+        *input_signal.values(),
+        theta=fir_filter["b"],
+        Utheta=fir_filter["Uab"],
+        kind=kind,
+    )
 
-    # define filter
-    b = np.array([0.01967691, -0.01714282, 0.03329653, -0.01714282, 0.01967691])
-    a = np.array([1.0])
+    assert np.allclose(y_fir, y_iir)
+    assert np.allclose(Uy_fir, Uy_iir)
 
-    # simulate input and output signals
-    Fs = 100e3  # sampling frequency (in Hz)
-    Ts = 1 / Fs  # sampling interval length (in s)
-    nx = 500
-    time = np.arange(nx) * Ts  # time values
 
-    # input signal + run methods
-    sigma_noise = 1e-2  # std for input signal
-    x = rect(time, 100 * Ts, 250 * Ts, 1.0, noise=sigma_noise)  # generate input signal
-    Ux = sigma_noise * np.ones_like(x)  # uncertainty/autocorrelation of input signal
-    Uab = np.diag(
-        np.zeros((len(a) + len(b) - 1))
-    )  # fully certain filter-parameters, otherwise FIR and IIR do not match! (see docstring of IIRuncFilter)
+def test_tf2ss(iir_filter):
+    """compare output of _tf2ss to (the very similar) scipy.signal.tf2ss"""
+    b = iir_filter["b"]
+    a = iir_filter["a"]
+    A1, B1, C1, D1 = _tf2ss(b, a)
+    A2, B2, C2, D2 = scipy.signal.tf2ss(b, a)
+
+    assert np.allclose(A1, A2[::-1, ::-1])
+    assert np.allclose(B1, B2[::-1, ::-1])
+    assert np.allclose(C1, C2[::-1, ::-1])
+    assert np.allclose(D1, D2[::-1, ::-1])
+
+
+def test_get_derivative_A():
+    """dA is sparse and only a specifc diagonal is of value -1.0"""
+    p = 10
+    dA = _get_derivative_A(p)
+    index1 = np.arange(p)
+    index2 = np.full(p, -1)
+    index3 = index1[::-1]
 
-    for kind in ["diag", "corr"]:
-        # run signal through both implementations
-        y_iir, Uy_iir, _ = IIRuncFilter(x, Ux, b, a, Uab=Uab, kind=kind)
-        y_fir, Uy_fir = FIRuncFilter(x, Ux, theta=b, Utheta=Uab, kind=kind)
+    sliced_diagonal = np.full(p, -1.0)
```

Comment:
What happens if we here do `index 2 = sliced_diagonal = np.full(p, -1.0)`. Would that be beneficial to clarify what's expected?